### ISSUE TITLE
Proposal: opt created time of image

### ISSUE THREAD

**Issue by zrss** [google/go-containerregistry#273]
## Expected

the image that generated by *ko* should has the correct created time rather than 48 years ago ...

## Additional info

i build a golang repo with *ko* in *Ubuntu 16.04.3 LTS*, but finally, when the build is completed, it shows that the created time of the image is 48 years ago (1970 ...)

https://github.com/google/go-containerregistry/blob/2f3e3e1a55fb70c98801f944d7b6249f8e8a3e20/cmd/ko/resolve.go#L33-L45

i propose if there is no valid `SOURCE_DATE_EPOCH` env, we can use the current time as the created time of the image

**Comment by zrss** [comment]
oh i see ~ but

> SOURCE_DATE_EPOCH=`date +%s` ko blabla ...

should be set before ko in `./hack/release.sh` ?

for example ...

https://github.com/knative/build/blob/58ec993bbc74f84f93c44c2234c188152f00143d/hack/release.sh#L55-L57

**Comment by ImJasonH** [comment]
Nope, that would have the effect of building and pushing a new image every day, whether or not the code itself changed, just because it was rebuilt on another day.

By producing a reproducible build, releases are faster and just as correct, and users who run the `:latest` image don't have to constantly pull the same image.

**Comment by zrss** [comment]
thanks @ImJasonH for a nice explaining

the key point is that created time should be avoid to producing a reproducible build, and if the code has not changed, it should not generate a new image

but ... 48 years ... maybe it has to clarify that we intend to do that, and *it is a feature not bug*
